pkg/analysis/numericbounds/analyzer.go

```diff
@@ -0,0 +1,216 @@
+var Analyzer = &analysis.Analyzer{
+	Name:     name,
+	Doc:      "Checks that numeric fields (int32, int64, float32, float64) have appropriate minimum and maximum bounds validation markers",
+	Run:      run,
+	Requires: []*analysis.Analyzer{inspector.Analyzer},
+}
+
+func run(pass *analysis.Pass) (any, error) {
+	inspect, ok := pass.ResultOf[inspector.Analyzer].(inspector.Inspector)
+	if !ok {
+		return nil, kalerrors.ErrCouldNotGetInspector
+	}
+
+	inspect.InspectFields(func(field *ast.Field, _ extractjsontags.FieldTagInfo, markersAccess markershelper.Markers) {
+		// Create TypeChecker with closure capturing markersAccess
+		typeChecker := utils.NewTypeChecker(func(pass *analysis.Pass, ident *ast.Ident, node ast.Node, prefix string) {
+			checkNumericType(pass, ident, node, prefix, markersAccess)
+		})
+
+		typeChecker.CheckNode(pass, field)
+	})
+
+	return nil, nil //nolint:nilnil
+}
+
+func checkNumericType(pass *analysis.Pass, ident *ast.Ident, node ast.Node, prefix string, markersAccess markershelper.Markers) {
+	// Only check int32, int64, float32, and float64 types
+	if ident.Name != "int32" && ident.Name != "int64" && ident.Name != "float32" && ident.Name != "float64" {
+		return
+	}
+
+	field, ok := node.(*ast.Field)
+	if !ok {
+		return
+	}
+
+	fieldMarkers := utils.TypeAwareMarkerCollectionForField(pass, markersAccess, field)
+
+	// Determine if this is a slice element (prefix contains "array element")
+	isSlice := containsArrayElement(prefix)
+
+	// Determine which markers to look for based on whether the field is a slice
+	minMarkers, maxMarkers := getMarkerNames(isSlice)
+
+	// Get minimum and maximum marker values
+	minimum, minErr := getMarkerNumericValue(fieldMarkers, minMarkers)
+	maximum, maxErr := getMarkerNumericValue(fieldMarkers, maxMarkers)
+
+	// Check if markers are missing
+	minMissing := errors.Is(minErr, errMarkerMissingValue)
+	maxMissing := errors.Is(maxErr, errMarkerMissingValue)
+
+	// Report any invalid marker values (e.g., non-numeric values)
+	if minErr != nil && !minMissing {
+		pass.Reportf(field.Pos(), "%s has an invalid minimum marker: %v", prefix, minErr)
+	}
+	if maxErr != nil && !maxMissing {
+		pass.Reportf(field.Pos(), "%s has an invalid maximum marker: %v", prefix, maxErr)
+	}
+
+	// Report if markers are missing
+	if minMissing {
+		pass.Reportf(field.Pos(), "%s is missing minimum bounds validation marker", prefix)
+	}
+	if maxMissing {
+		pass.Reportf(field.Pos(), "%s is missing maximum bounds validation marker", prefix)
+	}
+
+	// If any markers are missing or invalid, don't continue with bounds checks
+	if minErr != nil || maxErr != nil {
+		return
+	}
+
+	// Validate bounds are within the type's valid range
+	checkBoundsWithinTypeRange(pass, field, prefix, ident.Name, minimum, maximum)
+}
+
+// containsArrayElement checks if the prefix string contains "array element"
+// which indicates we're checking a slice element type.
+func containsArrayElement(prefix string) bool {
```

Comment:
You can use utilis.IsArrayType which is better than checking whether the prefix has specific strings.